src/clientiface.cpp

```diff
@@ -173,12 +173,16 @@ void RemoteClient::GetNextBlocks (
 	*/
 	s32 new_nearest_unsent_d = -1;
 
-	const s16 full_d_max = g_settings->getS16("max_block_send_distance");
-	const s16 d_opt = g_settings->getS16("block_send_optimize_distance");
+	s32 wanted_range = sao->getWantedRange() / MAP_BLOCKSIZE;
```

Comment:
How about an s16 since this is only being compared against s16 values? Since this is in units of blocks s32 is excessive and not needed.